File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    map<string, int> m;
    
    for(int i=0;i<clothes.size();i++) {
        m[clothes[i][1]]++;
    }
    
    int sum = 1;
    for(auto a : m) {
        sum *= (a.second + 1);
    }
    
    
    return sum-1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> m = new HashMap<>();
        for (String[] cloth : clothes) {
            String type = cloth[1];
            m.put(type, m.getOrDefault(type, 0) + 1);
        }
        
        int answer = 1;
        for (int count : m.values()) {
            answer *= (count+1);
        }
        return answer - 1;
    }
}